provider/vsphere/provider.go

```diff
@@ -67,11 +68,45 @@ func (p environProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+const failedLoginMsg = "ServerFaultCode: Cannot complete login due to an incorrect user name or password."
+
 // Ping tests the connection to the cloud, to verify the endpoint is valid.
 func (p environProvider) Ping(endpoint string) error {
-	// a stub for now until we get a real implementation.
-	_, err := url.Parse(endpoint)
-	return errors.Trace(err)
+	// try to be smart and not punish people for adding or forgetting http
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
+	}
+	switch u.Scheme {
+	case "http", "https":
+		// good!
+	case "":
+		u, err = url.Parse("https://" + endpoint + "/sdk")
+		if err != nil {
+			return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
+		}
+	default:
+		return errors.New("Invalid endpoint format, please use an http or https URL.")
+	}
+
+	client, logout, err := newConnection(u)
+	if err != nil {
+		return errors.Errorf("No VSphere server running at %s", endpoint)
+	}
+	defer logout()
+	err = client.Login(context.TODO(), nil)
+	if err == nil {
+		// shouldn't happen, but can't really complain if it does.
```

Comment:
The login process is a SOAP conversation.  I think it would be quite remarkable if you happened to connect to a machine that responded in the correct way to make us think we're logging into vsphere.  

But yes, I can make the explanation more clear.